### DISCUSSION TITLE
How to access Sentinel-2 L1C imagery from Earth Search AWS Element 84

### DISCUSSION THREAD

**Discussion by dbptbx** [gjoseph92/stackstac#197] [Q&A]
Hi @gjoseph92 ,

thank you in first place for this stackstac package. I'm intending to apply on getting Sentinel-2 L1C imagery. However, I get an 'Access Denied' error that I don't know how to solve. See below:
`[PATH-TO-PYTHON-ENV]/lib/python3.10/site-packages/stackstac/rio_reader.py:332: UserWarning: Error opening 's3://sentinel-s2-l1c/tiles/30/S/WG/2020/12/19/0/B12.jp2': RasterioIOError('Access Denied')`

Please, find attached a jupyter notebook (I added the .txt extension to be able to upload it) with my simple test so you can easily follow the process.
[sentinel2_read_stac_l1c.ipynb.txt](https://github.com/gjoseph92/stackstac/files/10373323/sentinel2_read_stac_l1c.ipynb.txt)

Thank you very much in advance for your help. Looking forward to your answer.
Best,
Diego

**Comment by TomAugspurger** [comment]
I think that the assets linked from https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l1c/items are in a requester-pays S3 Bucket. You would need to need to set your S3 credentials wherever you're reading the data: https://docs.aws.amazon.com/AmazonS3/latest/userguide/RequesterPaysExamples.html. Have you done that?

**Reply by dbptbx** [reply]
Thank you @TomAugspurger for suggestion. Actually, we have wrapped the call of stackstac.stack method with a rasterio environment:
```
with rio.Env(AWS_REQUEST_PAYER='requester'):
    data = stackstac.stack(items_s2, **stack_options)
```
and it just worked.

Best,

**Reply by gjoseph92** [reply]
xref https://github.com/gjoseph92/stackstac/issues/133

**Comment by dbptbx** [comment]
This issue is apparently solved here: https://github.com/cogeotiff/rio-tiler/issues/52

and here: https://github.com/rasterio/rasterio/issues/692
